src/crawl/DeepCrawler/DeepCrawler.js

```diff
@@ -0,0 +1,161 @@
+
+          // we're scraping the data from the listview itself, so just return the paginated pages themselves
+          if (pageInfo.hasAllFields || !pageInfo.detailViewUrlSelector || !pageInfo.detailViewUrlAttribute) {
+            for (const doc of paginatedDocs) {
+              yield Promise.resolve({ _url: doc.url });
+            }
+            return;
+          }
+
+          // otherwise, we need to grab the detail views
+
+          for (const doc of paginatedDocs) {
+            const $ = cheerio.load(doc.html);
+            const detailUrls = [];
+
+            $(pageInfo.detailViewUrlSelector).each((_, elem) => {
+              detailUrls.push(this.normalizeUrl($(elem).attr(pageInfo.detailViewUrlAttribute), url));
+            });
+
+            for (const it of detailUrls) {
+              yield Promise.resolve({ _url: it });
+            }
+          }
+
+          return;
+
+        case 'detail_view':
+          if (scrapeType === 'fetch_many') {
+            if (pageInfo.listViewUrl) {
+              urlStack.push(this.normalizeUrl(pageInfo.listViewUrl, url));
+              break;
+            }
+            // if it can't find the list view url, just continue to fetch the current page as a single entry
+          }
+          yield Promise.resolve({ _url: latestUrl });
+          return;
+
+        case 'unknown':
+          if (!pageInfo.guessUrl) return null;
+
+          urlStack.push(this.normalizeUrl(pageInfo.guessUrl, url));
+          break;
+      }
+    }
+  }
+
+  normalizeUrl(url, originalUrl) {
+    if (url.startsWith('http://') || url.startsWith('https://')) {
+      return url;
+    }
+    const origin = new URL(originalUrl).origin;
+    if (url.startsWith('//')) {
+      url = url.slice(1);
+    }
+    return new URL(url, origin).toString();
+  }
+
+  async analyzePage(html, userPrompt) {
+    let rawPrompt = ANALYZE_PAGE_PROMPT;
+    rawPrompt = rawPrompt.replace('{{html}}', html);
+    rawPrompt = rawPrompt.replace('{{prompt}}', userPrompt);
+
+    const { answer, truncatedPrompt } = await this.ai.ask(rawPrompt, {
+      format: 'json',
+    });
+
+    writeFileSync('logs/analyze-page-prompt.txt', truncatedPrompt);
+
+    return tryParseJson(answer);
+  }
+
+  async scrapePage(scrapeType, schema, html) {
+    let rawPrompt = SCRAPE_PAGE_PROMPT;
+    rawPrompt = rawPrompt.replace('{{scrapeType}}', scrapeType);
+    rawPrompt = rawPrompt.replace('{{schema}}', JSON.stringify(schema));
+    rawPrompt = rawPrompt.replace('{{html}}', html);
+
+    const { answer } = await this.ai.ask(rawPrompt, { format: 'json' });
+    try {
+      return tryParseJson(answer);
+    } catch (err) {
+      console.log('invalid response', answer);
+      throw err;
+    }
+  }
+
+  async getPromptScrapeType(prompt) {
+    const rawPrompt = SCRAPE_TYPE_PROMPT.replace('{{prompt}}', prompt);
+    const { answer } = await this.ai.ask(rawPrompt, { format: 'json' });
+    const parsed = tryParseJson(answer);
```

Comment:
`answer.partial` has the JSON, when you put in format: 'json' it will auto parse it